### PR TITLE
[aievec] Canonicalize transposed B contract op.

### PR DESCRIPTION
**Pull request by jsetoain**

### CODE REVIEW

**Comment by jsetoain** [comment]
@newling WIP on [Make loops-for-vectorization recognize matmul_transpose_b](https://github.com/nod-ai/iree-amd-aie/pull/331)

File: lib/Dialect/AIEVec/Transforms/VectorToVectorConversions.cpp [hunk 1]
```diff
@@ -39,6 +39,54 @@ using namespace xilinx::aievec;
...
+  auto lhsShape = op.getLhsType().getShape();
+  auto rhsShape = op.getRhsType().getShape();
+  auto accShape = cast<ShapedType>(op.getAccType()).getShape();
+  if (lhsShape.size() < 2 || rhsShape.size() < 2 || accShape.size() < 2)
+    return false;
+
+  // Check that the innermost iterators match gemm-like iterators
+  SmallVector<vector::IteratorType> iterators = op.getIteratorTypesArray();
+  auto innerMostIterators =
+      SmallVector<vector::IteratorType>(iterators.end() - 3, iterators.end());
```
[the remark this answers is not in the document]
Comment (jsetoain) [reply lib/Dialect/AIEVec/Transforms/VectorToVectorConversions.cpp:56]: > It's possible that `len(iterator_types)<3`?

I don't think so, no. At that point I've already checked that lhs/rhs/acc have at least 2 dimensions each, which implies at least 2 parallel dimensions (acc) and there has to be at least one contracting dimension.

File: lib/Dialect/AIEVec/Transforms/VectorToVectorConversions.cpp [hunk 2]
```diff
@@ -411,6 +460,107 @@ struct FlattenMultDimTransferWritePattern
   }
 };
 
+// This pattern takes out an implicit transposition of the `rhs` operand in a
```
Comment (david-vc) [inline lib/Dialect/AIEVec/Transforms/VectorToVectorConversions.cpp:463]: I suggest to clarify the comment as follows:
 an implicit transposition of the `rhs` operand
=>
 an implicit transposition of the 2 innermost dimensions of the `rhs` operand

**Review by david-vc — COMMENTED** [review]
Nice work.

**Pull request merged**